src/RPC/middleware.ts

```diff
@@ -8,6 +8,7 @@ import type {
 import { TransformStream } from 'stream/web';
 import * as rpcErrors from './errors';
 import * as rpcUtils from './utils';
+import { promise } from '../utils';
 const jsonStreamParsers = require('@streamparser/json');
 
 function binaryToJsonMessageStream<T extends JsonRpcMessage>(
```

Comment:
This function does not have a return type?

Reply:
Added a return type.